activerecord/lib/arel/visitors/mssql.rb

```diff
@@ -76,6 +76,15 @@ def visit_Arel_Nodes_SelectStatement(o, collector)
           end
         end
 
+        def visit_Arel_Nodes_SelectCore(o, collector)
+          collector = super
+          maybe_visit o.optimizer_hints, collector
+        end
+
+        def visit_Arel_Nodes_OptimizerHints(o, collector)
+          collector << "OPTION (#{sanitize_as_sql_comment(o).join(", ")})"
+        end
```

Comment:
Actually yes. I've just implemented the visitor method as an example for ibm_db, informix, mssql, since I can't find any 3rd party adapter for Rails 5.2 which uses the visitor.
(sqlserver adapter uses its own visitor https://github.com/rails-sqlserver/activerecord-sqlserver-adapter/blob/v5.2.0.rc2/lib/arel/visitors/sqlserver.rb)

We could improve the unsed visitors to prevent possible SQL injection, but there is no need to rush for me.